File: ws/leiton/U2/WS15/ZooApp/src/main/java/ec/edu/espe/zooapp/view/ZooApp.java
package ec.edu.espe.zooapp.view;

import ec.edu.espe.zooapp.model.Animal;
import ec.edu.espe.zooapp.model.Cage;
import ec.edu.espe.zooapp.model.Food;
import ec.edu.espe.zooapp.model.Mammal;
import ec.edu.espe.zooapp.model.Vaccine;
import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Jose Leiton
 */
public class ZooApp {

    public static void main(String[] args) {
        System.out.println("Hello World!");
        System.out.println("This is Jose's Zoo");
        
        Animal animal;
        int animalId;
        String name;
        Date bornOnDate;
        float weight;
        char sex;
        ArrayList<Vaccine> vaccines;
        int vaccineId;
        
        vaccineId=4;
        String vaccineName = "Paracetamol";
        
        animalId=1;
        name="Lucy";
        bornOnDate = new Date(2024, 0, 21);
        weight = 4.5F;
        sex='f';
        
        Vaccine vaccine;
        int VaccineId = 0;
        vaccine = new Vaccine(VaccineId, vaccineName);
        vaccines = new ArrayList<>();
        vaccines.add(vaccine);
        
        ArrayList<Animal>animals = new ArrayList<>();
        Cage cage;
        cage = new Cage(animalId, animals);
                
        animal = new Animal(animalId, name, bornOnDate, weight, sex, vaccines);
        System.out.println("My animal is -->"+animal);
        
        Mammal mammal = new Mammal(0.21F, 2, animalId, name, bornOnDate, weight, sex, vaccines);
        System.out.println("My mammal is -->"+mammal);
        
        vaccine = new Vaccine(5, "Ibuprofeno");
        
        mammal.breastFeed();
        mammal.assignCage(cage);
        mammal.feed(new Food());
        mammal.register();
        mammal.provideVaccination(vaccine);
        
        System.out.println("Mammal -->"+mammal);
        
        System.out.println("cage --> "+cage);
    }
}

File: hw/robalino/u2/Hw 18 Inheritance/zooApp.py
from datetime import date
from typing import List
from vaccine import Vaccine
from animal import Animal
from food import Food  
from cage import Cage
from mammal import Mammal




def main():
    print("Hello World!")
    print("This is Benjamin's Zoo")

    # Gather Animal data from user
    animal_id = int(input("Enter animal ID: "))
    name = input("Enter animal name: ")
    born_on_date = input("Enter the animal's birth date (YYYY-MM-DD): ")
    weight = float(input("Enter animal weight (in kg): "))
    sex = input("Enter animal sex (m/f): ").lower()

    # Convert born_on_date to a Python date object
    born_on_date = date.fromisoformat(born_on_date)

    # Create Vaccines
    vaccines: List[Vaccine] = []
    while True:
        add_vaccine = input("Do you want to add a vaccine? (yes/no): ").lower()
        if add_vaccine == "no":
            break
        try:
            vaccine_id = int(input("Enter vaccine ID: "))
            description = input("Enter vaccine description: ")
            vaccines.append(Vaccine(vaccine_id, description))
        except ValueError:
            print("Invalid vaccine ID. Please enter a numeric value.")

    # Create Animal
    animal = Animal(animal_id, name, born_on_date, weight, sex, vaccines)
    print(f"\nCreated Animal: {animal}\n")

    # Create Mammal
    maximum_hair_size = float(input("Enter mammal's maximum hair size (in cm): "))
    mammary_glands_number = int(input("Enter mammal's number of mammary glands: "))
    mammal = Mammal(maximum_hair_size, mammary_glands_number, animal_id, name, born_on_date, weight, sex, vaccines)
    print(f"\nCreated Mammal: {mammal}\n")

    # Create Cage and assign Mammal
    cage_id = int(input("Enter cage ID: "))
    cage = Cage(cage_id, [])
    mammal.assign_cage(cage)
    print(f"\nMammal assigned to Cage: {cage}\n")

    # Feed Mammal
    food_id = int(input("Enter food ID: "))
    food_amount = float(input("Enter food amount: "))
    food_type = input("Enter food type: ")
    food = Food(food_id, food_amount, food_type)
    mammal.feed(food)

    # Register Mammal
    mammal.register()

    # Provide additional Vaccine
    vaccine_id = int(input("Enter new vaccine ID: "))
    vaccine_description = input("Enter new vaccine description: ")
    new_vaccine = Vaccine(vaccine_id, vaccine_description)
    mammal.provide_vaccination(new_vaccine)

    # Final Output
    print("\nFinal Details:")
    print(f"Mammal: {mammal}")
    print(f"Cage: {cage}")


if __name__ == "__main__":
    main()